### PR TITLE
Find sequence for signed tx by bruteforce

### PR DESCRIPTION
**Pull request by webmaster128**
Implementation in `findSequenceForSignedTx`

### CODE REVIEW

**Comment by webmaster128** [comment]
Updated

File: packages/bcp/src/cosmwasmconnection.spec.ts [hunk 1]
```diff
@@ -330,18 +318,74 @@ describe("CosmWasmConnection", () => {
...
+      // Test properties of getTx result: height, transactionId, log, transaction
+      expect(height).toEqual(searchedHeight);
+      expect(transactionId).toEqual(searchedTransactionId);
+      assert(log, "Log must be available");
+      const [firstLog] = JSON.parse(log);
+      expect(firstLog.events.length).toEqual(2);
+      expect(transaction).toEqual(searchedTransaction);
+
+      // Signature test ensures the nonce is correct
+      expect(signatures.length).toEqual(1);
```
Comment (ethanfrey) [inline packages/bcp/src/cosmwasmconnection.spec.ts]: +1

File: packages/bcp/src/cosmwasmconnection.ts [hunk 2]
```diff
@@ -359,10 +366,32 @@ export class CosmWasmConnection implements BlockchainConnection {
...
+      types.isMsgInstantiateContract(firstMsg) ||
+      types.isMsgExecuteContract(firstMsg)
+    ) {
+      senderAddress = firstMsg.value.sender;
+    } else {
+      throw new Error(`Got unsupported type of message: ${firstMsg.type}`);
+    }
+
+    const { accountNumber, sequence: currentSequence } = await this.cosmWasmClient.getNonce(senderAddress);
+    const sequenceForTx = await findSequenceForSignedTx(
```
Comment (ethanfrey) [inline packages/bcp/src/cosmwasmconnection.ts]: The magic is here....

File: packages/sdk/src/cosmwasmclient.spec.ts [hunk 3]
```diff
@@ -187,6 +183,7 @@ describe("CosmWasmClient", () => {
         };
 
         const result = await client.postTx(marshalTx(signedTx));
+        await sleep(50); // wait until tx is indexed
```
Comment (ethanfrey) [inline packages/sdk/src/cosmwasmclient.spec.ts]: I really wish there was a way to do without sleep, but I didn't find it either with weave

File: packages/sdk/src/sequence.ts [hunk 4]
```diff
@@ -0,0 +1,46 @@
+import { Secp256k1, Secp256k1Signature, Sha256 } from "@iov/crypto";
+
+import { makeSignBytes } from "./encoding";
+import { decodeSignature } from "./signature";
+import { CosmosSdkTx } from "./types";
+
+/**
+ * Serach for sequence s with `min` <= `s` < `upperBound` to find the sequence that was used to sign the transaction
```
Comment (ethanfrey) [inline packages/sdk/src/sequence.ts]: Very nice docs.
And good algotithm

File: packages/sdk/src/testdata/txresponse1.json [hunk 5]
```diff
@@ -0,0 +1,57 @@
+{
```
Comment (ethanfrey) [inline packages/sdk/src/testdata/txresponse1.json]: Nice samples

Maybe you can add to the readme how they were generated, so more could be added if needed (I think I know how, just nice to doc)

**Comment by webmaster128** [comment]
:)

**Pull request merged**